src/interface/ui/accountactivity/BlockchainAccountActivity.cpp

```diff
@@ -150,30 +161,219 @@ auto BlockchainAccountActivity::display_balance(
     return blockchain::internal::Format(chain_, value);
 }
 
+enum class Diag {
+    load_thread,
+    portion_p,
+    delete_inactive,
+    emplace,
+    load_transaction_bitcoin,
+    process_txid,
+    blockchain_thread_item_id,
+    process_txid_lambda,
+    process_txid_custom,
+    add_item,
+    end
+};
+
+std::string to_string(Diag d);
+
+std::string to_string(Diag d)
+{
+    switch (d) {
+        case Diag::load_thread:
+            return "load_thread    ";
+        case Diag::portion_p:
+            return "portion_p    ";
+        case Diag::delete_inactive:
+            return "delete_inactive";
+        case Diag::emplace:
+            return "emplace        ";
+        case Diag::load_transaction_bitcoin:
+            return "lo.._bitcoin   ";
+        case Diag::process_txid:
+            return "process_txid   ";
+        case Diag::blockchain_thread_item_id:
+            return "bl.._item_id   ";
+        case Diag::process_txid_lambda:
+            return "pr.._lambda    ";
+        case Diag::process_txid_custom:
+            return "pr.._custom    ";
+        case Diag::add_item:
+            return "add_item       ";
+        case Diag::end:
+            return "end";
+        default:
+            return "???";
+    }
+}
+
 auto BlockchainAccountActivity::load_thread() noexcept -> void
 {
-    const auto transactions =
-        [&]() -> UnallocatedVector<blockchain::block::pTxid> {
-        try {
-            const auto& chain =
-                Widget::api_.Network().Blockchain().GetChain(chain_);
-            height_ = chain.HeaderOracle().BestChain().height_;
+    bool go{};
+    {
+        std::unique_lock<std::mutex> L(load_lock_);
+        if (load_in_progress_) {
+            tdiag("ZZZZ load_thread load_in_progress_");
```

Comment:
same as "RRRR":)

Reply:
Same reply :-)

Reply:
The ZZZZ tag is used in an ongoing investigation on how to speed up the initial processing in this module, so I'm inclined to keep it for a while.

Reply:
ok, good